### DISCUSSION TITLE
Display selected Day from the Calendar

### DISCUSSION THREAD

**Discussion by Leon910** [neuefische/questions#869] [Web] [answered]
Hi everyone,

I’m currently working on a US where I have a calendar day popup window that allows the user to get directed to the homepage and create a new emotion entry for that date. When the user clicks the "Create Emotion" button, it redirects to the home page, automatically opening the form where the selected date should be prefilled in the date and time input field.

The issue is: The selected date is only displayed in the form after I reload the page.

Here’s what I’ve implemented and tried so far:

- I pass the selected date as a query parameter (selectedDate) when redirecting to the form page.
- I use our already implemented useEffect to extract selectedDate from router.query and update a selectedDate state in the form component.
- I initialize the selectedDate state

[feature/calendar-popup](https://github.com/Jennifer-Hinrichsen/emotion-tracker/pull/83)

[US - calendar popup](https://github.com/users/Jennifer-Hinrichsen/projects/3/views/1?pane=issue&itemId=87622462&issue=Jennifer-Hinrichsen%7Cemotion-tracker%7C68)

Any help or suggestions would be greatly appreciated!

**Comment by onemanwenttomow** [comment, marked as answer]
I would consider checking if the `selectedDate` is in the query string when you initialise the component. 

i.e. something like this (if it doesn't exist on the query string then use the current date). 

``` js
const [selectedDate, setSelectedDate] = useState(() => {
  const currentDate = format(new Date(), "yyyy-MM-dd");
  return router.query.selectedDate || currentDate;
});
```

Hopefully this helps

**Reply by Leon910** [reply]
It works👌🏻🙌 Thanks a lot Peter - @onemanwenttomow !!!
